### DISCUSSION TITLE
Typescript errors don't break the build

### DISCUSSION THREAD

**Discussion by ayushdevelop** [gatsbyjs/gatsby#37524] [Help]
Hi folks, I am working on project where we are using `gatsbyjs` with `typescript`. 

I observed that typescript errors are not affecting gatsby build, which is not what we want, we want the gatsby build to fail if there are any typescript errors present in the repository  so that no bugs pass through into the production.

My question how can I make the gatsby build to fail if there is any unidentified typescript error present in the repository?

**Comment by LekoArts** [comment]
Gatsby uses Babel to support TypeScript which doesn't do typechecking. If you want to fail your build, run `tsc` before your actual build.
